### ISSUE TITLE
Add check on energy range

### ISSUE THREAD

**Issue by kialio** [fermiPy/fermipy#140]
If the emin value in the config is lower than the supplied ft1 file (i.e. if you grabbed data from the FSSC down to 100 MeV and you put emin: 60 in the config file) then you don't get an error until after the setup step (i.e. when using the fit, optimize or even the free_sources functions) and it is a bit of a cryptic error (see below).  I suggest putting a check when loading the config that the data file matches or have it throw a more specific error.

`
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-23-036db857f153> in <module>()
----> 1 gta.optimize(skip=names[1:])

/data/slag2/jsperki1/software/fermisoft/GlastExternals/python/2.7.13/lib/python2.7/site-packages/fermipy/gtanalysis.pyc in optimize(self, **kwargs)
   2052 
   2053         # Fix all parameters
-> 2054         self.free_sources(free=False, loglevel=logging.DEBUG)
   2055 
   2056         # Free norms of sources for which the sum of npred is a

/data/slag2/jsperki1/software/fermisoft/GlastExternals/python/2.7.13/lib/python2.7/site-packages/fermipy/gtanalysis.pyc in free_sources(self, free, pars, cuts, distance, skydir, minmax_ts, minmax_npred, exclude, square, **kwargs)
   1514 
   1515         for s in srcs:
-> 1516             self.free_source(s.name, free=free, pars=pars, **kwargs)
   1517 
   1518         return srcs

/data/slag2/jsperki1/software/fermisoft/GlastExternals/python/2.7.13/lib/python2.7/site-packages/fermipy/gtanalysis.pyc in free_source(self, name, free, pars, **kwargs)
   1750         # Deduce here the names of all parameters from the spectral type
   1751         src_par_names = pyLike.StringVector()
-> 1752         self.like[name].src.spectrum().getParamNames(src_par_names)
   1753 
   1754         par_indices = []

/data/slag2/jsperki1/software/fermisoft/Glast/v11r05p02-x86_64/python/SummedLikelihood.pyc in __getitem__(self, name)
    158             component.logLike.syncParams()
    159     def __getitem__(self, name):
--> 160         item = self.model[name]
    161         try:
    162             item.type

AttributeError: 'SummedLikelihood' object has no attribute 'model'
`

**Comment by woodmd** [comment]
I agree that it would be good to have some kind of consistency check on the inputs.  @terrijb suggested this a while back (see #32) but I just haven't had time to follow-up on it.  It's not completely trivial since it would involve writing some code to read FT1 files and parse the DSKEYs to figure out what data selections had been applied.  If you or someone else felt like working on this you'd be more than welcome.

**Comment by kialio** [comment]
Ok.  Just assigned to me.  I'll also look at #32 and maybe merge these issues.
